pkg/expansion/system.go

```diff
@@ -232,7 +228,20 @@ func expandResource(obj *unstructured.Unstructured, ns *corev1.Namespace, templa
 	resource := &unstructured.Unstructured{}
 	resource.SetUnstructuredContent(src)
 	resource.SetGroupVersionKind(resultantGVK)
-	resource.SetNamespace(ns.Name)
+	if ns != nil {
+		resource.SetNamespace(ns.Name)
+	} else {
+		nsFromUn, ok, err := unstructured.NestedString(obj.Object, "metadata", "namespace")
+		if err != nil {
+			return nil, fmt.Errorf("could not extract namespace field %q in resource %s", srcPath, obj.GetName())
+		}
+		if !ok {
```

Comment:
If namespace is missing, or nil, then we assume `obj` is a cluster-scoped object, and the generated resource is also cluster scoped (i.e. if `!ok`, then just don't call `resource.SetNamespace()`).

Stated simply: the resultant resource inherits the namespace from the source resource.

Reply:
Should we be returning error in this case then?

Reply:
Nope, just do not assign a namespace to the child, as a namespace was not assigned to the parent.

Reply:
good call out Max! Modified the setter to only apply if the parent has a namespace set.